homeassistant/data_entry_flow.py

```diff
@@ -371,6 +373,11 @@ async def _async_handle_step(
                 flow.flow_id, flow.handler, err.reason, err.description_placeholders
             )
 
+        # Setup the flow handler's preview if needed
+        if result.get("preview") is not None and flow.__class__ not in self._preview:
```

Comment:
can we use handler ?

Reply:
we should only check preview is in result if the step is a form. We don't support previews on other steps.

Reply:
Makes me wonder if we should add this check to `async_show_form` instead

Reply:
OK, I modified it a bit